src/lein_monolith/plugin.clj

```diff
@@ -4,76 +4,93 @@
   (:require
     [lein-monolith.config :as config]
     [lein-monolith.dependency :as dep]
+    [leiningen.core.eval :as eval]
     [leiningen.core.main :as lein]
     [leiningen.core.project :as project]))
 
-
 ;; ## Profile Generation
 
+(def profile-config
+  "Configuration for inherited profiles. Structured as a vector of pairs to
+  maintain ordering."
+  [[:monolith/inherited
+    {:types #{}
+     :ks {:inherit :inherit
+          :subproject :monolith/inherit}}]
+
+   [:monolith/inherited-raw
+    {:types #{:raw}
+     :ks {:inherit :inherit-raw
+          :subproject :monolith/inherit-raw}}]
+
+   [:monolith/leaky
+    {:types #{:leaky}
+     :ks {:inherit :inherit-leaky
+          :subproject :monolith/inherit-leaky}}]
+
+   [:monolith/leaky-raw
+    {:types #{:leaky :raw}
+     :ks {:inherit :inherit-leaky-raw
+          :subproject :monolith/inherit-leaky-raw}}]])
+
+
 (defn- subproject-dependencies
   "Given a map of internal projects, return a vector of dependency coordinates
   for the subprojects."
   [subprojects]
   (mapv #(vector (key %) (:version (val %))) subprojects))
 
 
-(defn- add-profile-paths
-  "Update a profile paths entry by adding the absolute paths from the given
-  project. Returns the updated profile."
-  [profile project k]
-  (update profile k into
-          (map (partial str (:root project) "/")
-               (get project k))))
+(defn- maybe-mark-leaky
+  "Add ^:leaky metadata to a profile if it is of the leaky type."
+  [profile types]
+  (cond-> profile
+    (:leaky types) (vary-meta assoc :leaky true)))
```

Comment:
Using `cond->` with a single test and expression is harder to read than a regular `if`, especially when you're not doing any transformation on the initial threaded value:
```suggestion
  (if (:leaky types)
    (vary-meta profile assoc :leaky true)
    profile))
```